components/cmake/build_model.cmake

```diff
@@ -259,7 +252,7 @@ function(build_model COMP_CLASS COMP_NAME)
     # Make sure we link blas/lapack
     target_link_libraries(${TARGET_NAME} BLAS::BLAS LAPACK::LAPACK)
 
-    set_target_properties(${TARGET_NAME} PROPERTIES LINKER_LANGUAGE ${LD})
+    set_target_properties(${TARGET_NAME} PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
Are you sure CXX is the right link language? I think the linker language used will impact the actual name of the program entry point (main), which may differ between CXX/Fortran. I think the proper way to do things is "use the language of the src file with the `main` program", and if you need cxx lib linked in, simply add it to the link line.

Reply:
I guess it's ok then. The common wisdom I've usually found online was to use the language of the main program, but I was never able to find an explanation of the rationale behind it.

Reply:
I was wrong, it looks like intel requires fortran `main` to be linked with fortran compilers.